patches/rust-libc/0001-managarm-initial-port.patch

```diff
@@ -0,0 +1,584 @@
++pub const SHUT_RD: ::c_int = 1;
++pub const SHUT_RDWR: ::c_int = 2;
++pub const SHUT_WR: ::c_int = 3;
++pub const SOCK_DGRAM: ::c_int = 1;
++pub const SOCK_STREAM: ::c_int = 4;
++pub const SOL_SOCKET: ::c_int = 1;
++pub const SO_BROADCAST: ::c_int = 6;
++pub const SO_ERROR: ::c_int = 5;
++pub const SO_RCVTIMEO: ::c_int = 11;
++pub const SO_REUSEADDR: ::c_int = 12;
++pub const SO_SNDTIMEO: ::c_int = 15;
++pub type sa_family_t = ::c_uint;
++s! {
++    pub struct sockaddr_storage {
++        pub ss_family: sa_family_t,
++        pub __padding: [c_char; 128 - ::mem::size_of::<sa_family_t>()],
++    }
++}
++
++// abis/mlibc/errno.h
++pub const EACCES: ::c_int = 1002;
++pub const EADDRINUSE: ::c_int = 1003;
++pub const EADDRNOTAVAIL: ::c_int = 1004;
++pub const EAGAIN: ::c_int = 1006;
++pub const EBADF: ::c_int = 1008;
++pub const ECONNABORTED: ::c_int = 1013;
++pub const ECONNREFUSED: ::c_int = 1014;
++pub const ECONNRESET: ::c_int = 1015;
++pub const EDEADLK: ::c_int = 1016;
++pub const EEXIST: ::c_int = 1019;
++pub const EINPROGRESS: ::c_int = 1024;
++pub const EINTR: ::c_int = 1025;
++pub const EINVAL: ::c_int = 1026;
++pub const ENOENT: ::c_int = 1043;
++pub const ENOSYS: ::c_int = 1051;
++pub const ENOTCONN: ::c_int = 1052;
++pub const EPERM: ::c_int = 1063;
++pub const EPIPE: ::c_int = 1064;
++pub const ERANGE: ::c_int = 3;
++pub const ETIMEDOUT: ::c_int = 1072;
++pub const EWOULDBLOCK: ::c_int = EAGAIN;
++
++// options/posix/include/fcntl.h
++pub const AT_FDCWD: ::c_int = -100;
++pub const F_DUPFD_CLOEXEC: ::c_int = 2;
++pub const F_GETFL: ::c_int = 5;
++pub const F_SETFL: ::c_int = 6;
++pub const O_ACCMODE: ::c_int = 7;
++pub const O_APPEND: ::c_int = 8;
++pub const O_CLOEXEC: ::c_int = 0x4000;
++pub const O_CREAT: ::c_int = 0x10;
++pub const O_EXCL: ::c_int = 0x40;
++pub const O_NONBLOCK: ::c_int = 0x400;
++pub const O_RDONLY: ::c_int = 2;
++pub const O_RDWR: ::c_int = 3;
++pub const O_TRUNC: ::c_int = 0x200;
++pub const O_WRONLY: ::c_int = 5;
++
++// options/mlibc/seek-whence.h
++pub const SEEK_CUR: ::c_int = 1;
++pub const SEEK_END: ::c_int = 2;
++pub const SEEK_SET: ::c_int = 3;
++
++// options/posix/include/netinet/tcp.h
++pub const TCP_NODELAY: ::c_int = 1;
++
++// abis/mlibc/stat.h
++pub const S_IFBLK: mode_t = 0x6000;
++pub const S_IFCHR: mode_t = 0x2000;
++pub const S_IFDIR: mode_t = 0x4000;
++pub const S_IFIFO: mode_t = 0x1000;
++pub const S_IFLNK: mode_t = 0xA000;
++pub const S_IFMT: mode_t = 0xF000;
++pub const S_IFREG: mode_t = 0x8000;
++pub const S_IFSOCK: mode_t = 0xC000;
++s! {
++    pub struct stat {
++        pub st_dev: dev_t,
++        pub st_ino: ino_t,
++        pub st_mode: mode_t,
++        pub st_nlink: nlink_t,
++        pub st_uid: uid_t,
++        pub st_gid: gid_t,
++        pub st_rdev: dev_t,
++        pub st_size: off_t,
++        // TODO: Should we use timespec here? Don't think we can
++        // pub st_atime: timespec,
++        // pub st_mtime: timespec,
++        // pub st_ctime: timespec,
```

Comment:
Other `std` ports use two members, not a single `timespec` member: https://docs.rs/libc/0.2.93/libc/struct.stat.html (so this comment can probably be deleted).

Reply:
Yeah, the unix-common code in `libc` relies on that struct being 'inlined' like that. But I wasn't sure if it would change where padding is placed. Seems like it should be fine though, will delete